firmware/Platypus.cpp

```diff
@@ -309,27 +299,55 @@ void Motor::onLoop_(void *data)
   self->loop();
 }
 
-Sensor::Sensor(int channel) 
-: channel_(channel)
+Sensor::Sensor(int id) : id_(id)
 {  
+  // Abstract Class - should not be instantiated
```

Comment:
Instead of this, can we add a pure virtual function to this class so it _has_ to be abstract?